lua/lspconfig/util.lua

```diff
@@ -298,6 +301,27 @@ function M.server_per_root_dir_manager(make_config)
         client_instance.workspace_folders = {}
       end
       table.insert(client_instance.workspace_folders, params.event.added[1])
+      if client_instance.config.new_folder_restart then
```

Comment:
this is a hidden property, used only by `lua/lspconfig/server_configurations/sumneko_lua.lua`. that is hard for other contributors to follow. 

where is the structure and valid fields of `client_instance.config` documented? if nvim-lspconfig is adding fields, then nvim-lspconfig should document this somewhere.

Reply:
this is a trouble thins about workspaceFolders . currently some very popular language server not support this. so if we want support workspaceFolders there has some work to do .. no idea currently maybe i can find a good way to do this. 
about lua server restart problem, I most suggest fixed in server not client ... so I will ask  sumneko for some help or create a pr for this in lua server repo.

Reply:
The issue has been fixed in https://github.com/sumneko/lua-language-server/commit/bddeb50920da81a6cecd91100aed018769bcc5b7

There's one more thing I'm confused about: why do we need to have a custom handler for `workspace/diagnostic/refresh` if you want to get rid of undefined global errors?

Can this also be fixed on the sumneko side instead?

Reply:
notice we don't have `workspace/diagnostic` handlers currently .  it's different `textDocument/diagnostic` .  

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_diagnostic

Reply:
True, I guess we should we have a handler for that? It's currently not causing any noticeable problems for other LSPs though.
I've also tried the handler snippet you posted, but what happens is that you get a flash of diagnostic errors before they're "fixed".
It feels ugly and incorrect, so I wonder whether there's a more correct way of handling it on lspconfig's side, or should a fix be made on sumneko's side?

Reply:
since the lua-language-server 3.6.7 released we don't need pr right ?

Reply:
Yes, we still need to do something about the handler though.